src/consumption.cpp

```diff
@@ -381,7 +381,7 @@ edible_rating player::can_eat( const item &food, bool interactive, bool force )
         overfull = !maybe_query( _( "You're full.  Force yourself to eat?" ) );
     } else if( ( ( nutr > 0 && temp_hunger < capacity ) ||
                  ( comest->quench > 0 && temp_thirst < capacity ) ) &&
-               !eathealth && !slimespawner ) {
+               !food.has_infinite_charges() && !eathealth && !slimespawner ) {
```

Comment:
Why is this relevant here at all? The character is going to eat only one charge anyway.